src/main.rs

```diff
@@ -68,10 +68,26 @@ fn main() {
 
     // Iterate all input elfs. Convert them to Tock friendly binaries and then
     // add them to the TAB file.
-    for elf_path in opt.input {
-        let tbf_path = elf_path.with_extension("tbf");
+    for elf_file in opt.input {
+        let (tbf_path, architecture) = if let Some(ref architecture) = elf_file.architecture {
```

Comment:
Why do we need `architecture`? We don't actually need to know the architecture for the rest of the elf2tab conversion.

Reply:
Why doesn't the old logic for 143 work?

Reply:
The  name of the file in the tab needs to be *architecture.tbf*. Libtock-rs uses a *target/tbf* folder where it places all the tbf and tab files, naming them with application_name.{tbf,tab}. This change will make elf2tab name the tbf file with application.architecture.tbf, which is different from what the tab file expects.

As a side note, it will require a change in libtock-c, as the elf files there are named using the architecture. I can add a check here to see if the application name is the same with the architecture (libtock-c's case).

Reply:
Why is elf2tab changing the name of the elf when creating the tbf? Decoupling the architecture from the filename makes sense to me, but re-introducing new dependencies between elf name and tbf name seems like it has the opposite effect. For example, a build system might reasonably use the name `application-architecture.elf` which would then become `application-architecture.architecture.elf`. I think we should not add the extra name change and just use the same name as what elf2tab is given (with a different extension).

Reply:
I see your point. I'll change it back. 

As a note, this will override applications in libtock-rs if an app is compiled for multiple architectures.

Reply:
> As a note, this will override applications in libtock-rs if an app is compiled for multiple architectures.

Right, if the same elf is used for multiple architectures. That could be a problem (although I think it is separate from this one). I think we should revisit this if we want to support one elf -> multiple architectures, but right now I think either we should a) let the compiler take advantage of a specific architecture and _not_ share elfs across different versions of an arch (preferred), or b) make this a tockloader (or other loader tool) problem, and have TABs contain only unique TBFs but allow tockloader to use a more generic arch if the specific arch doesn't exist.

Reply:
I think a) is easier for now. I'll revert the changes. In the long term, I think it might be worth discussing in detail how we should change tockloader to have a better fix than this.